capacity-mode-evaluator/src/pricing.py

```diff
@@ -17,8 +17,9 @@ def __init__(self, region_name):
         # pick the closest endpoint to the supplied region
         if region_name not in AMERICAN_REGIONS:
             closest_api_region = 'ap-south-1'
-
-        self.pricing_client = boto3.client(
+        
+        self.session = boto3.session.Session(profile_name=profile_name)
```

Comment:
Implement the necessary conditional and modify the code to accommodate None as an input for customers utilizing environment variables.

Reply:
Updated to `None`